pkg/lwip/include/sock_types.h

```diff
@@ -90,6 +90,9 @@ struct lwip_sock_base {
  */
 struct sock_ip {
     lwip_sock_base_t base;          /**< parent class */
+    mutex_t mutex;                  /**< Mutex to protect the sock */
+    struct netbuf *last_buf;        /**< Last received data */
+    bool peek_buf_avail;            /**< Whether there is a buffer left from a previous peek operation */
```

Comment:
Should those be inside an 

```C
#ifdef MODULE_SOCK_AUX_PEEK
…
#endif
```
?

Reply:
That would unfortunately entail `ifdef`s all over, since `IS_USED(MODULE_SOCK_AUX_PEEK)` is not enough for the application to compile (tons of `has no member named`). This would clutter the code too much. In my opinion the trade-off that these variables are always included vs cluttering the code with `ifdef`s is ok, but I can understand that people here might disagree.

Reply:
But it will add 16 bytes to each socket, even when not used.

Reply:
> But it will add 16 bytes to each socket, even when not used.

I added a fixup commit showing how it would look like when removing the members from the struct instead. If you prefer this I'll merge this into the branch.

> Btw: Why is the mutex needed now? Can sockets now be shared between multiple threads?

Now that you mention it... Seems like they are not intended to be thread-safe at all. I removed the mutex entirely.